### ISSUE TITLE
Error: Unknown key "inliner"

### ISSUE THREAD

**Issue by anajuliabit** [ethereum/solidity#15576]
## Description

solc doesn't recognize `inliner` option

## Environment

- Compiler version: `Version: 0.8.19+commit.7dd6d404.Darwin.appleclan`
- Target EVM version (as per compiler settings): `shanghai`
- Framework/IDE (e.g. Truffle or Remix): foundry
- EVM execution environment / backend / blockchain client:
- Operating system: macOS (Apple Silicon)

## Steps to Reproduce

Compilation with IR active fails with 

``
Error: Compiler run failed:
Error: Unknown key "inline"
``

For reference and how to reproduce https://github.com/foundry-rs/foundry/issues/9322

**Comment by zerosnacks** [comment]
For additional context / recap:

Steps to reproduce

1. Clone https://github.com/moonwell-fi/moonwell-contracts-v2
2. Run `forge coverage --ir-minimum --skip test --skip script --out artifacts/coverage`

This yields: `Error: Compiler run failed: Error: Unknown key "inline"`

When you run: `forge coverage --ir-minimum`

it yields: 

```
Error: solc exited with signal: 11 (SIGSEGV) (core dumped)
<empty output>
```

**Comment by cameel** [comment]
Just from a quick look, this could be caused by [`settings.optimizer.details.inliner` setting](https://docs.soliditylang.org/en/develop/using-the-compiler.html#input-description) being set in the wrong dict. If that's the case that would be a problem on Foundry side. On the other hand `solc` should not segfault because of it so if that's the case, it's something we'd still need to fix.

It would be helpful if we could remove some layers of indirection and reproduce this directly with solc. For that we'd need to see what is the exact Standard JSON content and command-line flags that forge is passing to solc.

**Comment by grandizzy** [comment]
> Just from a quick look, this could be caused by [`settings.optimizer.details.inliner` setting](https://docs.soliditylang.org/en/develop/using-the-compiler.html#input-description) being set in the wrong dict.

Indeed, that's the case, we're setting `settings.optimizer.details.inliner` to false even for solc versions <= 0.8.4 (`Standard JSON: Properly allow the inliner setting under settings.optimizer.details` was addressed in 0.8.5). We can just omit it from details, can you pls confirm that the default is set to false for versions >= 0.8.5? (in this way we could support older versions too without an explicit check). ([foundry ref](https://github.com/foundry-rs/foundry/issues/9322#issuecomment-2480484349)) Thanks!

**Comment by cameel** [comment]
I can confirm that the defaults have not changed ever since the feature was introduced in 0.8.2 (#10761). But note that the default is dependent on the state of the optimizer: `true` when the optimizer is enabled and `false` otherwise.

What changed in 0.8.5 (#11419) is just that you can actually use the setting in Standard JSON - there was a bug in validation logic that prevented it from being used. But it did not affect the defaults.
